apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientResourceProvider.java

```diff
@@ -353,31 +375,74 @@ private CcwCodebookVariable partDCwVariableFor(String system) {
    * @param joins to add for many-to-one relations
    * @return the criteria
    */
-  private CriteriaQuery<Beneficiary> queryBeneficiariesBy(
+  private TypedQuery<Beneficiary> queryBeneficiariesBy(
       SingularAttribute<Beneficiary, String> field,
       String value,
       PatientLinkBuilder paging,
       List<SetAttribute<Beneficiary, ?>> joins) {
-
     CriteriaBuilder cb = entityManager.getCriteriaBuilder();
     CriteriaQuery<Beneficiary> criteria = cb.createQuery(Beneficiary.class);
     Root<Beneficiary> b = criteria.from(Beneficiary.class);
+
     joins.forEach(f -> b.fetch(f, JoinType.LEFT));
 
-    if (paging.isPagingRequested() && !paging.isFirstPage()) {
-      return criteria
-          .select(b)
-          .where(
-              cb.and(
-                  cb.equal(b.get(field), value),
-                  cb.greaterThan(b.get("beneficiaryId"), paging.getCursor())))
-          .orderBy(cb.asc(b.get("beneficiaryId")));
-    } else {
-      return criteria
-          .select(b)
-          .where(cb.equal(b.get(field), value))
-          .orderBy(cb.asc(b.get("beneficiaryId")));
-    }
+    Predicate whereClause =
+        paging.isPagingRequested() && !paging.isFirstPage()
+            ? cb.and(
+                cb.equal(b.get(field), value),
+                cb.greaterThan(b.get("beneficiaryId"), paging.getCursor()))
+            : cb.equal(b.get(field), value);
+
+    return entityManager.createQuery(
+        criteria.select(b).where(whereClause).orderBy(cb.asc(b.get("beneficiaryId"))));
+  }
+
+  /**
+   * Build a criteria for a general beneficiaryId query
+   *
+   * @param field to match on
+   * @param value to match on
+   * @param paging to use for the result set
+   * @return the criteria
+   */
+  private TypedQuery<String> queryBeneficiaryIds(
+      SingularAttribute<Beneficiary, String> field, String value, PatientLinkBuilder paging) {
+    CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+    CriteriaQuery<String> criteria = cb.createQuery(String.class);
+    Root<Beneficiary> b = criteria.from(Beneficiary.class);
+
+    Predicate whereClause =
```

Comment:
Although this whereClause is similar to the previous whereClause in the previous method. It is not. I tried to build a common function and it didn't work.